task/bq2bq/main.go

```diff
@@ -389,16 +389,27 @@ func (b *BQ2BQ) GenerateDependencies(ctx context.Context, request models.Generat
 		return nil, errors.New("empty sql file")
 	}
 
+	selfTable, err := b.GenerateDestination(ctx, models.GenerateDestinationRequest{
+		Config:  request.Config,
+		Assets:  request.Assets,
+		Project: request.Project,
+	})
+	if err != nil {
+		return response, err
+	}
+
 	// first parse sql statement to find dependencies and ignored tables
-	parsedDependencies, ignoredDependencies, err := b.FindDependenciesWithRegex(ctx, request)
+	parsedDependencies, ignoredDependencies, err := b.FindDependenciesWithRegex(ctx, request, selfTable.Destination)
 	if err != nil {
 		return response, err
 	}
 
 	// try to resolve referenced tables directly from BQ APIs
 	response.Dependencies, err = b.FindDependenciesWithRetryableDryRun(timeoutCtx, queryData.Value, svcAcc)
 	if err != nil {
-		return response, err
+		if !strings.Contains(err.Error(), fmt.Sprintf("Not found: Table %s was not found", selfTable.Destination)) {
```

Comment:
can you add a comment here as well stating why we are doing this?

Reply:
Sure, added the comments